flink-core/src/test/java/org/apache/flink/api/common/io/BinaryInputFormatTest.java

```diff
@@ -160,6 +160,44 @@ public void testGetStatisticsMultiplePaths() throws IOException {
                 stats.getTotalInputSize());
     }
 
+    @Test
+    public void testCreateInputSplitsWithEmptySplit() throws IOException {
+        // create temporary file with 3 blocks
+        final File tempFile = File.createTempFile("binary_input_format_test", "tmp");
+        tempFile.deleteOnExit();
```

Comment:
The `tempFile` can be deleted only if the whole JVM exits. I would suggest to use `TemporaryFolder` which can be cleaned after the test method finishes. `FlinkUserCodeClassLoadersTest` can be an example to reference.

Reply:
fixed